swc/swc.bzl

```diff
@@ -0,0 +1,64 @@
+"swc rule"
+
+load("//swc/private:swc.bzl", _swc_lib = "swc")
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+_swc = rule(
+    implementation = _swc_lib.implementation,
+    attrs = _swc_lib.attrs,
+    toolchains = _swc_lib.toolchains,
+)
+
+_SUPPORTED_EXTENSIONS = [".ts", ".tsx", ".jsx", ".mjs", ".cjs"]
```

Comment:
in theory we can transpile `js` to `js` but it should only be allowed if the output is re-rooted so output file name .doesn't collide with input.

leave a "TODO: support .js with output re-rooting"?

Reply:
yup. One nice thing about the way this is documented is that advanced users can always eject to `genrule` if this rule is too limiting